homeassistant/components/ituran/sensor.py

```diff
@@ -0,0 +1,119 @@
+"""Sensors for Ituran vehicles."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime
+
+from pyituran import Vehicle
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    DEGREE,
+    UnitOfElectricPotential,
+    UnitOfLength,
+    UnitOfSpeed,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import IturanConfigEntry
+from .coordinator import IturanDataUpdateCoordinator
+from .entity import IturanBaseEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class IturanSensorEntityDescription(SensorEntityDescription):
+    """Describes Ituran sensor entity."""
+
+    value_fn: Callable[[Vehicle], StateType | datetime]
+
+
+SENSOR_TYPES: list[IturanSensorEntityDescription] = [
+    IturanSensorEntityDescription(
+        key="address",
```

Comment:
The address and heading ones I'm a bit sceptical about tbh. What would you automate with those that the device tracker is not already giving you?

Reply:
We often tend to include things just because we can. Even though they are disabled by default imho those add so little to none value that we might just leave them out. If you wanna automate with address chances are, you want to now if your car is in XYZ and then a zone would be required for automation anyway and if your heading somewhere also needs to happen over a continous amount of time so the proximity integration is the place to be.

Reply:
I'm having some problems letting go of this one. Especially of the address which can be useful for the UI, not only for automations.
If it's OK with you, I'll leave this unresolved for now and wait for input from the via team and whatever's decided then I'll do.

Reply:
discussed that and it's ok if you think it's useful

Reply:
Thanks!